tests/link/jax/test_nlinalg.py

```diff
@@ -82,7 +82,7 @@ def assert_fn(x, y):
 
 @pytest.mark.xfail(
     version_parse(jax.__version__) >= version_parse("0.2.12"),
-    reason="Omnistaging cannot be disabled",
+    reason="JAX Numpy API does not support dynamic shapes",
```

Comment:
Why does this test fail? Also seems like it should just work?

Reply:
I was looking at it. It actually failed, but not due to dynamic shape error with omnistaging. For this one, when I removed the `xfail` to see the underlying error, I got `jax.errors.TracerIntegerConversionError: The __index__() method was called on traced array with shape int64[].` instead. Checking [here](https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerIntegerConversionError) I see that the error is due to passing in a traced value in place of a Python integer. To get rid of it, we need to mark the value at the function level. That is another PR I guess.

Reply:
Hmm can you try to pass `dvector(shape=(2,))` and see if it works? Since that's not the concern of the test it's a shame to have it xfail

Reply:
PyTensor is explicitly not compatible with numpy 2.0 at the moment: https://github.com/pymc-devs/pytensor/issues/688

Reply:
`dvector()` has no `shape` argument so I tried `vector(dtype="float64", shape=(2,))` instead. The error persists. The stack trace pointed the error to start from this temporary function generated when we turn the `Op` to JAX:
```py
def jax_funcified_fgraph(tensor_variable):
    # MaxAndArgmax{axis=(0,)}(<Vector(float64, shape=(2,))>)
    tensor_variable_1, argmax = maxandargmax(tensor_variable)
    # Mul(max, argmax)
    tensor_variable_2 = elemwise_fn(tensor_variable_1, argmax)
    return (tensor_variable_2,)
```
The solution from [here](https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerIntegerConversionError) suggests that we need to somehow mark `tensor_variable` in the function above as static with `functools.partial` and `jax.jit`, but I think that it is out of scope of this PR? Maybe we can mark this as xfail for "Operation leads to JAX Tracer object is used in a context where a Python integer is expected (jax.errors.TracerIntegerConversionError)."